src/System.IO.FileSystem/tests/Directory/Modify_FailSafe.cs

```diff
@@ -0,0 +1,581 @@
+        fe.MoveNext();
+        if (fe.Name.Equals(".") && ((fe.Attributes & FileAttributes.Directory) != 0))
+            fe.MoveNext();
+        if (!fe.Name.Equals(".."))
+            throw new Exception("Trying to skip . and .. in FileEnumerator test - not what I expected  "+fe.Name);
+        int num = 0;
+        Console.WriteLine("\tFileEnumerator Creation, Last Write, and Last Access Times and Size:");
+        while(fe.MoveNext()) {
+            if (!fe.Name.Equals(files[num]))
+                throw new Exception("FileEnumerator name wasn't what was expected!  got: "+fe.Name);
+            Console.WriteLine("\t"+fe.CreationTime+"\t"+fe.LastWriteTime+"\t"+fe.LastAccessTime+"  "+fe.Size);
+            num++;
+        }
+        if (num != files.Length)
+            throw new Exception("Files.Length wasn't equal number of files enumerated!  got: "+num+"  expected: "+files.Length);
+        if (fe.MoveNext())
+            throw new Exception("FileEnumerator GetNext returned true after enumeration finished!");
+        fe.Close();
+        */
+        Directory.SetCurrentDirectory(curDir);
+
+        // Delete files.
+        for (int i = 0; i < s_files.Length; i++)
+        {
+            File.Delete(dirName + "/" + s_files[i]);
+        }
+
+        dir.Delete();
+
+        Assert.True(s_pass);
+    }
+
+    public static void CreateTestFiles()
+    {
+        for (int i = 0; i < s_files.Length; i++)
+        {
+            Stream fs = File.Create(s_files[i]);
+            BinaryWriter bw = new BinaryWriter(fs);
+            bw.Write(s_files[i]);
+            bw.Dispose();
+        }
+    }
+
+    [Fact]
+    public static void CreateDirTest()
+    {
+        String dir = "CreateDirTest";
+
+        try
+        {
+            Directory.Delete(dir);
+        }
+        catch (IOException) { }
+
+        Directory.CreateDirectory(dir);
+
+
+        Directory.Delete(dir);
+
+        Assert.True(s_pass);
+    }
+
+    [Fact]
+    public static void DeleteDirTest()
+    {
+        String dir = "DeleteDirTest";
+
+        try
+        {
+            Directory.CreateDirectory(dir);
+        }
+        catch (IOException)
+        {
+        }
+
+
+        try
+        {
+            Directory.Delete(dir);
+            Stream f = File.Create(dir);
+            f.Dispose();
+            File.Delete(dir);
+        }
+        catch (IOException io)
+        {
+            Console.WriteLine("File name: \"" + dir + '\"');
+            Console.WriteLine("caught IOException when trying to delete the dir then open file");
+            Console.WriteLine(io);
+            Console.WriteLine(io.StackTrace);
+        }
+        catch (Exception e)
+        {
+            Console.Error.WriteLine("Got unexpected exception from Delete or surrounding code: " + e);
+            Console.WriteLine(e.StackTrace);
+            throw;
+        }
+
+        Assert.True(s_pass);
+    }
+
+    [Fact]
+    [ActiveIssue(0)] // SetCurrentDirectory
```

Comment:
Need something here other than just ActiveIssue(0).  Either file an issue or delete the test.